<launch>
  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="false"/>
  <arg name="gui" default="false"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/> 
  <arg name="verbose" default="false"/>

  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <!--include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find robocon25_sim)/worlds/robocon25.world"/>
    <arg name="verbose" 	value="true"/>
    <arg name="paused"		value="false"/>
  </include-->
  
  <!-- Spawn a robot into Gazebo -->
  <!--node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-file $(find robocon25_sim)/models/robot.model -urdf -model robot" /-->

  <!-- 红球放置位置 -->
  <!--arg name="red_ball_x" default="1.0"/-->
  <!--node name="red_ball" pkg="gazebo_ros" type="spawn_model" args="-file $(find wpr_simulation)/models/balls/red_ball.model -x $(arg red_ball_x) -urdf -model red_ball" /-->

  <!-- 启动距离标定节点 -->
  <node name="calibrate_dist" pkg="robocon_localization" type="calibrate_dist" output="screen"/>

  <!-- 调用usb摄像头 -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node">
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="camera_link" />
    <param name="autowhitebalance" value="true" />
    <param name="io_method" value="mmap"/>
    <!-- <remap from="/usb_cam/image_raw" to="/omni_camera/image_raw"/> -->
  </node>

</launch>


